sdk/Lusid.Sdk.Tests/tutorials/Instruments/DemoCreditDefaultSwap.cs

```diff
@@ -0,0 +1,157 @@
+                dayCountConvention: "Act365",
+                paymentCalendars: new List<string>(),
+                resetCalendars: new List<string>(),
+                rollFrequency: "6M",
+                settleDays: 2,
+                resetDays: 2
+            );
+            
+            var cdsProtectionDetailSpecification = new CdsProtectionDetailSpecification(
+                seniority: CdsProtectionDetailSpecification.SeniorityEnum.SNR,
+                restructuringType: CdsProtectionDetailSpecification.RestructuringTypeEnum.CR,
+                protectStartDay: true,
+                payAccruedInterestOnDefault: false);
+
+            var cds = new CreditDefaultSwap(
+                ticker: "ACME",
+                startDate: new DateTimeOffset(2020, 2, 7, 0, 0, 0, TimeSpan.Zero),
+                maturityDate: new DateTimeOffset(2020, 9, 18, 0, 0, 0, TimeSpan.Zero),
+                flowConventions: cdsFlowConventions,
+                couponRate: 0.5m,
+                protectionDetailSpecification: cdsProtectionDetailSpecification,
+                instrumentType: LusidInstrument.InstrumentTypeEnum.CreditDefaultSwap
+            );
+            // ASSERT that it was created
+            Assert.That(cds, Is.Not.Null);
+
+            // CAN NOW UPSERT TO LUSID
+            string uniqueId = Guid.NewGuid().ToString();
+            UpsertOtcInstrumentToLusid(cds,  uniqueId);
+
+            // CAN NOW QUERY FROM LUSID
+            var retrieved = QueryOtcInstrumentFromLusid(uniqueId);
+            Assert.That(retrieved.InstrumentType == LusidInstrument.InstrumentTypeEnum.CreditDefaultSwap);
+            var roundTripCds = retrieved as CreditDefaultSwap;
+            Assert.That(roundTripCds, Is.Not.Null);
+            Assert.That(roundTripCds.CouponRate, Is.EqualTo(cds.CouponRate));
+            Assert.That(roundTripCds.Ticker, Is.EqualTo(cds.Ticker));
+            Assert.That(roundTripCds.MaturityDate, Is.EqualTo(cds.MaturityDate));
+            Assert.That(roundTripCds.StartDate, Is.EqualTo(cds.StartDate));
+            Assert.That(roundTripCds.FlowConventions.Currency, Is.EqualTo(cds.FlowConventions.Currency));
+            Assert.That(roundTripCds.FlowConventions.PaymentFrequency, Is.EqualTo(cds.FlowConventions.PaymentFrequency));
+            Assert.That(roundTripCds.FlowConventions.ResetDays, Is.EqualTo(cds.FlowConventions.ResetDays));
+            Assert.That(roundTripCds.FlowConventions.SettleDays, Is.EqualTo(cds.FlowConventions.SettleDays));
+            Assert.That(roundTripCds.FlowConventions.PaymentCalendars.Count, Is.EqualTo(cds.FlowConventions.PaymentCalendars.Count));
+            Assert.That(roundTripCds.FlowConventions.PaymentCalendars, Is.EquivalentTo(cds.FlowConventions.PaymentCalendars));
+            
+            DeleteItems(null,null,null,uniqueId);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void DemoCreditDefaultSwapValuation(bool inlineValuation)
```

Comment:
what is the purpose of the inline vs non-inline valuations? or put another way, how does a reader know (or care) when to use one vs the other?

Reply:
For quick lite pricing of an instrument inline valuations should be used (no need to construct a whole portfolio just to get a price of an option), but showing how to do non-inline will give a better understanding of how lusid works (also show how to value a "structured instrument" e.g. strip of options represented in portfolio). Demonstrating the two side by side aims to avoid confusion (only showing one will mean users could be surprised by seeing another totally different way at some other point).

Reply:
i wonder if we just show the non-inline version given that how they get booked can impact how the instrument is created e.g. if the instrument definition should be unitized or not e.g. notationals?